app/scripts/lib/util.js

```diff
@@ -139,6 +139,17 @@ const addHexPrefix = (str) => {
   return `0x${str}`;
 };
 
+const fixHexValue = (str) => {
+  const prefixed = addHexPrefix(str);
+  // if it's a valid hex return it immediately
+  if (prefixed.match(/^-?0x[a-fA-F0-9]+$/u)) {
+    return prefixed;
+  }
+
+  // if we get here, throw invalid hex value detected error
+  throw new Error(`Invalid hex value ${str} detected.`);
```

Comment:
We should return one of the standardized Ethereum JSON-RPC errors here instead, like we do for the other error cases in `validateTxParams`. It should be ` ethErrors.rpc.invalidParams` in this case I think.